com.regnosys.rosetta.lib/src/main/java/com/rosetta/model/lib/expression/ExpressionOperators.java

```diff
@@ -262,4 +268,30 @@ private static <T> String formatMultiError(Mapper<T> o) {
 				t.getClass().getSimpleName() : // for rosettaModelObjects only log class name otherwise error messages are way too long
 				o.getMulti().toString();
 	}
+	
+	// one-of and choice
+
+	public static <T> ComparisonResult choice(Mapper<T> mapper, List<String> choiceFieldNames, ValidationResult.ChoiceRuleValidationMethod necessity) {
```

Comment:
I added a single static method to `ExpressionOperators`. Didn't remove anything, so 100% backwards compatible.

Reply:
Not sure I understand why we need to reflectively create this type.
If we need it for backward compatibility, then we should refactor it out into `BackwardCompatibilityChoice` type.

Reply:
This one is not for backwards compatibility. It's the runtime support for the "new" operations `one-of` and `optional/required choice`.

An expression `optional choice foo, bar`, which is syntactic sugar for `item optional choice foo, bar`, will generate more or less the following code.
```
choice(item, Lists.of("foo", "bar"), ChoiceRuleValidationMethod.OPTIONAL)
```

There are a couple of alternatives to using reflection here, but I'm not sure what the best option would be:
1. instead of representing the attributes as `String`s, we could represent them as lambdas, i.e., `Lists.of(a -> a.getFoo(), a -> a.getBar())`. This is definitely possible, but a bit more tricky to get right.
2. we could generate a runtime representation of attributes in every `RosettaModelObject` class. Would require a bit more work.
3. we could implement it in the way that `only exists` works by passing in the evaluated attributes directly. However, the way that it works in `only exists` is terrible: it will evaluate parent object again for each attribute. For `only exists` this is actually okay because parent expressions are restricted to simple feature calls. That's not the case here though.

So... I think (1) is better, (2) is the best but more work, and (3) is terrible. How I've done it now is just the simple way.